### PR TITLE
Tao 1013 adaptive choice

### PR DESCRIPTION
**Pull request by hutnikau**

### CODE REVIEW

File: views/js/pciCreator/dev/adaptiveChoiceInteraction/test/commonRenderer.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                        relBaseUrl : 'qtiItemPci/pciCreator/dev/adaptiveChoiceInteraction/test/samples/',
+                        runtimeLocations : {
+                            adaptiveChoiceInteraction : 'qtiItemPci/pciCreator/dev/adaptiveChoiceInteraction/'
+                        },
+                        assertions : {
+                            interactionCount : 1,
+                            responses : []
+                        }
+                    });
+
```
Comment (ssipasseuth) [inline views/js/pciCreator/dev/adaptiveChoiceInteraction/test/commonRenderer.html]: Unfortunately, that level of unit testing we used to have is no longer sufficient. Now, we need all the functionnalities, possible events, get/setResponse, get/setState  to be fully tested, at least for the renderer of new interactions.
You can have a more up-to-date example of test in here : https://github.com/oat-sa/extension-tao-itemqti/blob/master/views/js/test/qtiCommonRenderer/interactions/choice/test.js

**Comment by krampstudio** [comment]
Please move your tests to `views/js/test`

File: views/js/pciCreator/dev/adaptiveChoiceInteraction/runtime/adaptiveChoiceInteraction.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        destroy : function () {
+            this.$container.off().empty();
+        },
+        /**
+         * Restore the state of the interaction from the serializedState.
+         * 
+         * @param {Object} interaction
+         * @param {Object} serializedState - json format
+         */
+        setSerializedState : function (state) {
```
Comment (krampstudio) [inline views/js/pciCreator/dev/adaptiveChoiceInteraction/runtime/adaptiveChoiceInteraction.js]: You should implement the state management (even though they're only wrapper to set/getResponse)

File: views/js/pciCreator/dev/adaptiveChoiceInteraction/runtime/js/renderer.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+ * Copyright (c) 2015 (original work) Open Assessment Technologies;
+ *               
+ */   
+define([
+    'ui/feedback',
+    'i18n'
+],
+function (feedback, __) {
+    'use strict';
+    return function (options) {
```
Comment (krampstudio) [inline views/js/pciCreator/dev/adaptiveChoiceInteraction/runtime/js/renderer.js]: This pattern isn't something we recommend. 
I'd prefer you to create a factory, like : 

```javascript 
return function adaptiveChoiceRenderer(options) {
      //...
     return {
         render : function render(){
             ///...
         }
      };
};
```

**Comment by hutnikau** [comment]
Can be reviewed.

**Comment by dietertaotesting** [comment]
Tom will get in touch with you and explain what is still missing. Mind you that I won't be here until Aug 3rd so you would probably need to assign the new PR to Sam.

**Pull request closed without merging**